sympy/solvers/ode/ode.py

```diff
@@ -1231,7 +1231,7 @@ class in it.  Note that a hint may do this anyway if
                     _, u = mul.as_independent(x, f(x))
                     break
             if u and u.has(f(x)):
-                h = x**(degree(Poly(u.subs(f(x), y), gen=x)))*f(x)
+                h = x**(degree(Poly(u.subs(f(x), y)), gen=x))*f(x)
```

Comment:
@oscarbenjamin does this look ok?

Reply:
You can remove `Poly` - 
`h = x**(degree(u.subs(f(x), y), gen=x))*f(x)`

Reply:
> You can remove `Poly` -
> `h = x**(degree(u.subs(f(x), y), gen=x))*f(x)`

then it fails for system of odes tests

Reply:
Which test does it fail for? It should be possible to remove Poly. Does the output of degree change if the call to Poly is removed?

Reply:
Maybe tests are failing due to the fractional/variables(a, b type) powers in the expression as without making `Poly(expression)`, calling degree is failing for this type of equations.
e.g-
```julia
In [174]: factor                                                                                     
Out[174]: 
 3/2
x   

In [175]: degree(factor.subs(f(x), y), gen=x)                                                        
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
~/Documents/git/sympy/sympy/polys/polyutils.py in _parallel_dict_from_expr_if_gens(exprs, opt)
    210 
--> 211                         monom[indices[base]] = exp
    212                     except KeyError:

KeyError: sqrt(x)

During handling of the above exception, another exception occurred:

PolynomialError                           Traceback (most recent call last)
<ipython-input-175-bfd99a930e51> in <module>
----> 1 degree(factor.subs(f(x), y), gen=x)

~/Documents/git/sympy/sympy/polys/polytools.py in degree(f, gen)
   4445                 p, _ = poly_from_expr(f)
   4446             else:
-> 4447                 p, _ = poly_from_expr(f, gen)
   4448 
   4449     if isNum:

~/Documents/git/sympy/sympy/polys/polytools.py in poly_from_expr(expr, *gens, **args)
   4252     """Construct a polynomial from an expression. """
   4253     opt = options.build_options(gens, args)
-> 4254     return _poly_from_expr(expr, opt)
   4255 
   4256 

~/Documents/git/sympy/sympy/polys/polytools.py in _poly_from_expr(expr, opt)
   4274         expr = expr.expand()
   4275 
-> 4276     rep, opt = _dict_from_expr(expr, opt)
   4277     if not opt.gens:
   4278         raise PolificationFailed(opt, orig, expr)

~/Documents/git/sympy/sympy/polys/polyutils.py in _dict_from_expr(expr, opt)
    366 
    367     if opt.gens:
--> 368         rep, gens = _dict_from_expr_if_gens(expr, opt)
    369     else:
    370         rep, gens = _dict_from_expr_no_gens(expr, opt)

~/Documents/git/sympy/sympy/polys/polyutils.py in _dict_from_expr_if_gens(expr, opt)
    305 def _dict_from_expr_if_gens(expr, opt):
    306     """Transform an expression into a multinomial form given generators. """
--> 307     (poly,), gens = _parallel_dict_from_expr_if_gens((expr,), opt)
    308     return poly, gens
    309 

~/Documents/git/sympy/sympy/polys/polyutils.py in _parallel_dict_from_expr_if_gens(exprs, opt)
    215                         else:
    216                             raise PolynomialError("%s contains an element of "
--> 217                                                   "the set of generators." % factor)
    218 
    219             monom = tuple(monom)

PolynomialError: x**(3/2) contains an element of the set of generators.
```

Same behavior for `x**a`.

Reply:
Oh yes, that's true. However the degree is inaccurate in those cases when Poly is used e.g.:
```julia
In [6]: degree(Poly(sqrt(x)), gen=x)                                                                                              
Out[6]: 0

In [7]: degree(Poly(sqrt(x), gen=x))                                                                                              
Out[7]: 1

In [8]: degree(sqrt(x))                                                                                                           
Out[8]: 1

In [9]: degree(sqrt(x), gen=x)                                                                                                    
---------------------------------------------------------------------------
...
PolynomialError: sqrt(x) contains an element of the set of generators.
```
The degree of `sqrt(x)` as a polynomial in `x` does not really make sense so I guess the last version that raises is the one we want although perhaps the exception should be caught.

Reply:
I have opened a pr to fix `reduced_separable` method.

Reply:
okay so i am reverting this change in ode.py back